src/main/java/org/wildfly/security/x500/X500PrincipalUtil.java

```diff
@@ -192,8 +200,14 @@ public static X500Principal asX500Principal(Principal principal, boolean convert
         if (principal instanceof X500Principal) {
             return (X500Principal) principal;
         }
-        if (HAS_X500_NAME && principal instanceof X500Name) {
-            return ((X500Name) principal).asX500Principal();
+        if (X500_NAME_CLASS != null && X500_NAME_CLASS.isAssignableFrom(principal.getClass())) {
+            try {
+                return (X500Principal) AS_X500_PRINCIPAL_HANLDE.invoke(principal);
+            } catch(RuntimeException | Error ex) {
+                throw ex;
+            } catch(final Throwable t) {
+                throw new RuntimeException(t);
```

Comment:
An undeclared exception type should always be wrapped with UndeclaredThrowableException.